server/src/main/java/org/elasticsearch/cluster/routing/package-info.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * <p>Cluster routing package provides routing information and manages shard
+ * allocations in cluster. The routing part contains two different views on
+ * shards in cluster {@link org.elasticsearch.cluster.routing.RoutingTable} and
+ * {@link org.elasticsearch.cluster.routing.RoutingNodes}. RoutingTable provides
+ * view from index to shard to node, and RoutingNodes provides view from node to
+ * shard. Shard allocation is a process of assigning and moving shards between
+ * nodes. For more details about allocation see {@link
+ * org.elasticsearch.cluster.routing.allocation}.</p>
+ *
+ * <b>Routing Table</b>
```

Comment:
RoutingTable, RoutingNodes and reroute, they have own documentation. It wasnt my intention to replace or duplicate what existing docs say, but rather point out how they interact which each other. It might be good to have some of these notes duplicated at class/method level. May be next PR